source/analyze-code.txt

```diff
@@ -0,0 +1,179 @@
+
+A builder is a class provided by the {+driver-short+} to help you construct
+`BSON <https://bsonspec.org/>`__ objects.
+
+LINQ is a query syntax included in the C# language. The {+driver-short+}
+can translate a subset of LINQ expressions into MongoDB aggregation pipelines.
+
+To learn more about builders, see 
+`Builders <{+driver-docs+}reference/driver/definitions/>`__ in the
+{+driver-short+} documentation.
+
+To learn more about LINQ, see the following resources:
+
+- `LINQ <{+driver-docs+}reference/driver/crud/linq/>`__ in the {+driver-short+}
+  documentation
+- `LINQ <https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/>`__
+  in the Microsoft C# guide
+
+To learn more about aggregation pipelines, see
+:manual:`Aggregation </aggregation/>` in the MongoDB manual.
+
+Analyze Builders
+----------------
+
+Use the {+product+} to translate your builder expressions into the {+query-api+}.
+Click the following tabs to see an example of a builder expression
+and its corresponding {+query-api+} translation:
+
+.. tabs::
+
+   .. tab:: Builders
+      :tabid: builder
+
+      .. code-block:: csharp
+
+         var filter = Builders<Movie>.Filter.Eq(m => m.Genre, genre) &
+                      Builders<Movie>.Filter.Gte(m => m.Score, minScore) &
+                      Builders<Movie>.Filter.Regex(m => m.Score, titleSearchTerm);
+
+   .. tab:: {+query-api+}
+      :tabid: query-api-builders
+
+      .. code-block:: json
+
+         {
+           "$and": [ { "Genre": genre },
+             { "Score": { "$gte": minScore } },
+             { "Score": /titleSearchTerm/ } ]
+         }
+
+Analyze Builders in Visual Studio
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To analyze your builder expressions in Visual Studio, perform the following actions:
+
+1. Install the {+product+} as described in the :doc:`Install </install>`
+   guide.
+2. Write a builder expression with the {+driver-short+}
+3. Move your mouse over the :guilabel:`...` annotation beneath the first
+   method of your builder expression to display an information message that contains
+   the {+query-api+} translation.
+
+Click the following tabs to see a builder expression with and without the
+information message displayed:
+
+.. tabs::
+
+   .. tab:: No Message
+      :tabid: no-message
+
+      .. figure:: /includes/images/builder.png
+         :alt: Screenshot of builder expression in visual studio with elipsis annotation.
+
+   .. tab:: Message
+      :tabid: message
+
+      .. figure:: /includes/images/builder-popup-photoshop.png
+         :alt: Screenshot of builder expression in visual studio with information message displayed.
+
+.. include:: /includes/command-line.rst
+
+Analyze LINQ
+------------
+
+Use the {+product+} to learn the following
+about your LINQ expressions:
+
+- How your LINQ expressions translate into the {+query-api+}
+- If your LINQ expressions contain an unsupported operator
```

Comment:
question: Is there a reference table of unsupported operators?

Reply:
After looking around for it, I am unable to find such a list. I'll ask Boris if he is aware of one.

Reply:
Unfortunately there no such list.
The closest thing would be our tests (which does not contains all examples yet):
[Unsupported Linq](https://github.com/BorisDog/mongo-csharp-analyzer/blob/vs20/tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressionsBasicTestCases.cs)
[Unsupported Linq2 but supported in Linq3](https://github.com/BorisDog/mongo-csharp-analyzer/blob/vs20/tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinq2TestCases.cs)
[Unsupported Builders](https://github.com/BorisDog/mongo-csharp-analyzer/blob/vs20/tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/NotSupportedBuildersExpressions.cs)